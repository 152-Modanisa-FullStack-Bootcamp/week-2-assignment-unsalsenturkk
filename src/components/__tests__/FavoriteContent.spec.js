import {shallowMount} from "@vue/test-utils";
import favoriteContent from "@/components/FavoriteContent";

describe("FavoriteContent.vue",()=>{

    it('sanity test ',  () => {
        return
    });

    describe("Favorite unit test",()=>{
        describe("exists check", () => {
            it('should FavoriteContent component exists', () => {

                const video = {
                    "id": 1,
                    "videoAddress": "https://www.youtube.com/watch?v=FXpIoQ_rT_c",
                    "coverImage": "https://raw.githubusercontent.com/modanisa/bootcamp-video-db/main/video-images/1-cover.webp",
                    "hoverImage": "https://raw.githubusercontent.com/modanisa/bootcamp-video-db/main/video-images/1-hover.webp",
                    "title": "Vue.js Course for Beginners [2021 Tutorial]",
                    "viewCount": 254,
                    "publishDateInMonth": 4,
                    "ownerImage": "https://yt3.ggpht.com/ytc/AKedOLTtJvQ1Vfew91vemeLaLdhjOwGx3tTBLlreK_QUyA=s68-c-k-c0x00ffffff-no-rj",
                    "ownerName": "freeCodeCamp.org",
                    "description": "Learn Vue 3 by in this full course. Vue.js is an open-source model–view–view model front end JavaScript framework for building user interfaces and single-page applications."
                }

                const wrapper = shallowMount(favoriteContent, {
                    propsData: {
                        video
                    }
                })

                expect(wrapper.exists()).toBeTruthy()
            });
            it('should render favorite-img', () => {

                const video = {
                    "id": 1,
                    "videoAddress": "https://www.youtube.com/watch?v=FXpIoQ_rT_c",
                    "coverImage": "https://raw.githubusercontent.com/modanisa/bootcamp-video-db/main/video-images/1-cover.webp",
                    "hoverImage": "https://raw.githubusercontent.com/modanisa/bootcamp-video-db/main/video-images/1-hover.webp",
                    "title": "Vue.js Course for Beginners [2021 Tutorial]",
                    "viewCount": 254,
                    "publishDateInMonth": 4,
                    "ownerImage": "https://yt3.ggpht.com/ytc/AKedOLTtJvQ1Vfew91vemeLaLdhjOwGx3tTBLlreK_QUyA=s68-c-k-c0x00ffffff-no-rj",
                    "ownerName": "freeCodeCamp.org",
                    "description": "Learn Vue 3 by in this full course. Vue.js is an open-source model–view–view model front end JavaScript framework for building user interfaces and single-page applications."
                }

                const wrapper = shallowMount(favoriteContent, {
                    propsData: {
                        video
                    }
                })

                const favorite_img = wrapper.find("#favorite-img")
                expect(favorite_img.exists()).toBeTruthy()

            })
            it('should render favorite-details', () => {
                const video = {
                    "id": 1,
                    "videoAddress": "https://www.youtube.com/watch?v=FXpIoQ_rT_c",
                    "coverImage": "https://raw.githubusercontent.com/modanisa/bootcamp-video-db/main/video-images/1-cover.webp",
                    "hoverImage": "https://raw.githubusercontent.com/modanisa/bootcamp-video-db/main/video-images/1-hover.webp",
                    "title": "Vue.js Course for Beginners [2021 Tutorial]",
                    "viewCount": 254,
                    "publishDateInMonth": 4,
                    "ownerImage": "https://yt3.ggpht.com/ytc/AKedOLTtJvQ1Vfew91vemeLaLdhjOwGx3tTBLlreK_QUyA=s68-c-k-c0x00ffffff-no-rj",
                    "ownerName": "freeCodeCamp.org",
                    "description": "Learn Vue 3 by in this full course. Vue.js is an open-source model–view–view model front end JavaScript framework for building user interfaces and single-page applications."
                }

                const wrapper = shallowMount(favoriteContent, {
                    propsData: {
                        video
                    }
                })

                const favorite_details = wrapper.find("#favorite-details")
                expect(favorite_details.exists()).toBeTruthy()
            })
        })
        describe('check functionality', () => {
            it("check displayHoverImg functionality", () => {
                const localThis = {
                    hoverImage: false,
                }
                favoriteContent.methods.displayHoverImg.call(localThis)
                expect(localThis.hoverImage).toBeTruthy()
            })
            it("check displayCoverImg functionality", () => {
                const localThis = {
                    hoverImage: false,
                }
                favoriteContent.methods.displayCoverImg.call(localThis)
                expect(localThis.hoverImage).toBeFalsy()
            })
            it("check navigateToFavoritePage functionality", () => {
                const video = {
                    "id": 1,
                    "videoAddress": "https://www.youtube.com/watch?v=FXpIoQ_rT_c",
                    "coverImage": "https://raw.githubusercontent.com/modanisa/bootcamp-video-db/main/video-images/1-cover.webp",
                    "hoverImage": "https://raw.githubusercontent.com/modanisa/bootcamp-video-db/main/video-images/1-hover.webp",
                    "title": "Vue.js Course for Beginners [2021 Tutorial]",
                    "viewCount": 254,
                    "publishDateInMonth": 4,
                    "ownerImage": "https://yt3.ggpht.com/ytc/AKedOLTtJvQ1Vfew91vemeLaLdhjOwGx3tTBLlreK_QUyA=s68-c-k-c0x00ffffff-no-rj",
                    "ownerName": "freeCodeCamp.org",
                    "description": "Learn Vue 3 by in this full course. Vue.js is an open-source model–view–view model front end JavaScript framework for building user interfaces and single-page applications."
                }
                const localThis = {
                    video,
                    $router: {
                        push: jest.fn()
                    },
                }
                favoriteContent.methods.navigateToFavoritePage.call(localThis)
                expect(localThis.$router.push).toBeCalled()
            })
        })
    })
    describe("Favorite snapshot test",()=>{
        it('should renders FavoriteContent correctly', () => {

            const video = {
                "id": 1,
                "videoAddress": "https://www.youtube.com/watch?v=FXpIoQ_rT_c",
                "coverImage": "https://raw.githubusercontent.com/modanisa/bootcamp-video-db/main/video-images/1-cover.webp",
                "hoverImage": "https://raw.githubusercontent.com/modanisa/bootcamp-video-db/main/video-images/1-hover.webp",
                "title": "Vue.js Course for Beginners [2021 Tutorial]",
                "viewCount": 254,
                "publishDateInMonth": 4,
                "ownerImage": "https://yt3.ggpht.com/ytc/AKedOLTtJvQ1Vfew91vemeLaLdhjOwGx3tTBLlreK_QUyA=s68-c-k-c0x00ffffff-no-rj",
                "ownerName": "freeCodeCamp.org",
                "description": "Learn Vue 3 by in this full course. Vue.js is an open-source model–view–view model front end JavaScript framework for building user interfaces and single-page applications."
            }

            const wrapper = shallowMount(favoriteContent, {
                propsData: {
                    video
                }
            })

            expect(wrapper.element).toMatchSnapshot()
        });
    })
})