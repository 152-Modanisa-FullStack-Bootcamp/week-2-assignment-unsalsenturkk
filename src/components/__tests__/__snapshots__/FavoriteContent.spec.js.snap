// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteContent.vue Favorite snapshot test should renders FavoriteContent correctly 1`] = `
<div
  class="m-flex
           m-flex-row"
  id="favorite-content"
>
  <div
    class="m-flex
               m-flex-column"
    id="favorite-img"
  >
    <img
      alt="Vue.js Course for Beginners [2021 Tutorial]"
      src="https://raw.githubusercontent.com/modanisa/bootcamp-video-db/main/video-images/1-cover.webp"
    />
     
    <img
      alt="Vue.js Course for Beginners [2021 Tutorial]"
      src="https://raw.githubusercontent.com/modanisa/bootcamp-video-db/main/video-images/1-hover.webp"
      style="display: none;"
    />
  </div>
   
  <div
    class="m-flex
             m-flex-column
             m-justify-content-space-around"
    id="favorite-details"
  >
    <div
      class="m-flex"
      id="title"
    >
      
      Vue.js Course for Beginners [2021 Tutorial]
    
    </div>
     
    <div
      class="m-flex
              m-flex-column"
      id="meta"
    >
      <div
        class="m-flex"
        id="metadata-line"
      >
        <span
          id="viewCount"
        >
          
           254 B görüntüleme
         
        </span>
         
        <span
          id="separator"
        >
          
            •
          
        </span>
         
        <span
          id="publishDateInMonth"
        >
          
            4 ay önce
         
        </span>
      </div>
    </div>
     
    <div
      class="m-flex"
      id="owner"
    >
      <img
        id="avatar"
        src="https://yt3.ggpht.com/ytc/AKedOLTtJvQ1Vfew91vemeLaLdhjOwGx3tTBLlreK_QUyA=s68-c-k-c0x00ffffff-no-rj"
      />
       
      <div
        id="ownerName"
      >
        
        freeCodeCamp.org
      
      </div>
    </div>
     
    <div
      class="m-flex"
      id="description"
    >
      
      Learn Vue 3 by in this full course. Vue.js is an open-source model–view–view model front end JavaScript framework for building user interfaces and single-page applications.
    
    </div>
  </div>
</div>
`;
